overlord/install/install.go

```diff
@@ -193,3 +206,232 @@ func CheckEncryptionSupport(model *asserts.Model, tpmMode secboot.TPMProvisionMo
 
 	return res.Type, res.UnavailableErr
 }
+
+// BuildInstallObserver creates an observer for gadget assets if
+// applicable, otherwise the returned gadget.ContentObserver is nil.
+// The observer if any is also returned as non-nil trustedObserver if
+// encryption is in use.
+func BuildInstallObserver(model *asserts.Model, gadgetDir string, useEncryption bool) (
+	observer gadget.ContentObserver, trustedObserver *boot.TrustedAssetsInstallObserver, err error) {
+
+	// observer will be a nil interface by default
+	trustedObserver, err = boot.TrustedAssetsInstallObserverForModel(model, gadgetDir, useEncryption)
+	if err != nil && err != boot.ErrObserverNotApplicable {
+		return nil, nil, fmt.Errorf("cannot setup asset install observer: %v", err)
+	}
+	if err == nil {
+		observer = trustedObserver
+		if !useEncryption {
+			// there will be no key sealing, so past the
+			// installation pass no other methods need to be called
+			trustedObserver = nil
```

Comment:
not here, because trustedObserver itself has a pointer type, in fact the strange return signature of this function was designed around that, it's definitely strange, stranger as an exported function vs local helper but I didn't want to redesign it on the fly, maybe we should leave a comment about reworking this at some point but it might get absorbed again by larger helpers doing more at some point